src/query/mlt/mlt.rs

```diff
@@ -0,0 +1,387 @@
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            FieldType::I64(_) => {
+                for field_value in field_values {
+                    let val = field_value
+                        .value()
+                        .i64_value()
+                        .ok_or(TantivyError::InvalidArgument("invalid value".to_string()))?;
+                    if !self.is_noise_word(val.to_string()) {
+                        let term = Term::from_field_i64(field, val);
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            FieldType::F64(_) => {
+                for field_value in field_values {
+                    let val = field_value
+                        .value()
+                        .f64_value()
+                        .ok_or(TantivyError::InvalidArgument("invalid value".to_string()))?;
+                    if !self.is_noise_word(val.to_string()) {
+                        let term = Term::from_field_f64(field, val);
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+
+    /// Determines if the term is likely to be of interest based on "more-like-this" settings
+    fn is_noise_word(&self, word: String) -> bool {
+        let word_length = word.len();
+        if word_length == 0 {
+            return true;
+        }
+        if self
+            .min_word_length
+            .map(|min| word_length < min)
+            .unwrap_or(false)
+        {
+            return true;
+        }
+        if self
+            .max_word_length
+            .map(|max| word_length > max)
+            .unwrap_or(false)
+        {
+            return true;
+        }
+        return self.stop_words.contains(&word);
+    }
+
+    /// Couputes the score for each term while ignoring not useful terms
+    fn create_score_term(
+        &self,
+        searcher: &Searcher,
+        per_field_term_frequencies: HashMap<Term, usize>,
+    ) -> Result<Vec<ScoreTerm>> {
+        let mut score_terms = BinaryHeap::new();
+        let num_docs = searcher
+            .segment_readers()
+            .iter()
+            .map(|x| x.num_docs() as u64)
+            .sum::<u64>();
+
+        for (term, term_frequency) in per_field_term_frequencies.into_iter() {
+            // ignore terms with less than min_term_frequency
+            if self
+                .min_term_frequency
+                .map(|x| term_frequency < x)
+                .unwrap_or(false)
+            {
+                continue;
+            }
+
+            let doc_freq = searcher.doc_freq(&term)?;
+
+            // ignore terms with less than min_doc_frequency
+            if self
+                .min_doc_frequency
+                .map(|x| doc_freq < x)
```

Comment:
no x here.

Reply:
I recommend call self.min_doc_frequency -> self.min_doc_frequency_opt, and using min_doc_frequency as the temp variable name.

Reply:
I really don't this this a good style. it's kind of redondant to embed the type knowledge in the variable name. That is fine for dynamically typed languages but here we already have that in the type (i.e Option<T>).
This is why `shadowing` was added IHMO but you also advised to not use shadowing. 
I also cannot call the temp variable `min_doc_frequency`. because it is also used in the context of `max_doc_frequency`.
In this case I think it's ok to shadow if the variable's scope does not  span multiple lines
```rust 
if self
                .min_doc_frequency
                .map(|min_doc_frequency| doc_freq < min_doc_frequency)
                .unwrap_or(false)
            {
                continue;
            }
```

Reply:
Let me start by saying I am ok with your solution.

> I really don't this this a good style.

es I think my opinion on shadowing is not common in the rust world.

> it's kind of redondant to embed the type knowledge in the variable name.
> That is fine for dynamically typed languages but here we already have that in the type (i.e Option).

Well.. yes and no. It is not as if the type is often visible on the screen is it?
If it is inferred by the compiler, then it is visible nowhere. (well in your editor hint if you  are in a powerful editor)
If it is in your struct definition, it is 3 screens above what you are reading. 

The way I see it, redundancy is not good in programming for two reasons...
- it increases signal to noise ratio. Here it is so minor I think it does not apply much
- it increases the number of places to write when doing a modification. Were we to refactor the variable 
to an enum, we would have to rename the variable name.
I'm ok with the redundancy of adding _opt on that kind of variable.

> This is why shadowing was added IHMO but you also advised to not use shadowing.

Interestingly shadowing was added without being a conscious choice, and now it is just there. 
https://internals.rust-lang.org/t/history-of-shadowing-in-rust/6441. 
Most rust developers like that feature.